FlowCryptUI/Nodes/PublicKeyDetailNode.swift

```diff
@@ -0,0 +1,162 @@
+        node.accessibilityIdentifier = "aid-public-key-label"
+        return node
+    }()
+
+    private lazy var fingerprintNode: ASTextNode = {
+        let node = ASTextNode()
+        node.attributedText = "fingerprint_label_value"
+            .localizeWithArguments(input.fingerprint.spaced(every: 4))
+            .attributed(Constants.fontStyle)
+        node.accessibilityIdentifier = "aid-fingerprint-value"
+        return node
+    }()
+
+    private lazy var warningLabel: ASTextNode = {
+        let node = ASTextNode()
+        node.attributedText = "public_key_import_warning"
+            .localized
+            .attributed(Constants.fontStyle, color: .warningColor)
+        node.accessibilityIdentifier = "aid-warning-label"
+        return node
+    }()
+
+    private lazy var toggleNode: SwitchCellNode = {
+        let input = SwitchCellNode.Input(
+            isOn: false,
+            attributedText: "show_public_key"
+                .localized
+                .attributed(Constants.fontStyle, color: .textColor),
+            accessibilityIdentifier: "aid-toggle-public-key-node",
+            backgroundColor: .clear,
+            switchJustifyContent: .center
+        )
+        let node = SwitchCellNode(input: input) { isOn in
+            self.shouldDisplayPublicKey = isOn
+        }
+        return node
+    }()
+
+    private lazy var publicKeyValueNode: ASTextNode = {
+        let node = ASTextNode()
+        node.attributedText = input.publicKey.attributed(Constants.fontStyle, color: .main)
+        node.accessibilityIdentifier = "aid-public-key-value"
+        return node
+    }()
+
+    private lazy var importKeyButtonNode: ASButtonNode = {
+        let node = ASButtonNode()
+        var title: String
+        var isEnabled = true
+        var bgColor: UIColor = .warningColor
+
+        switch input.importStatus {
+        case .alreadyImported:
+            title = "already_imported"
+            isEnabled = false
+            bgColor = .gray
+        case .differentKeyImported:
+            title = "update_public_key"
+        case .notYetImported:
+            title = "import_public_key"
+        }
+
+        node.setTitle(title.localized, with: .boldSystemFont(ofSize: 16), with: .white, for: .normal)
+        node.accessibilityIdentifier = "aid-import-key-button"
+        node.addTarget(self, action: #selector(importKeyButtonTapped), forControlEvents: .touchUpInside)
+
+        node.isEnabled = isEnabled
+        node.backgroundColor = bgColor
+        node.style.flexGrow = 1.0
+        node.style.flexShrink = 1.0
+        node.style.preferredSize.height = 40
+
+        return node
+    }()
+
+    var shouldDisplayPublicKey = false {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+
+    public var onImportKey: (() -> Void)?
+
+    let input: Input
+
+    public init(input: Input) {
+        self.input = input
+        super.init()
+        backgroundColor = UIColor(hex: "FAFAFA")
```

Comment:
Need to change color for dark mode:

<img width="200" alt="Screenshot 2023-09-13 at 15 47 00" src="https://github.com/FlowCrypt/flowcrypt-ios/assets/1062093/9d286a65-e71c-416e-93f3-18e93b3827b7">

Reply:
Nice catch 👍  Let me fix